### PR TITLE
DRA: E2E Node: test GRPC timeout

### PR DESCRIPTION
**Pull request by bart0sh**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Test GRPC timeout by slowing down NodePrepareResource call on the test plugin side.

#### Special notes for your reviewer:

This test case was implemented when trying to reproduce https://github.com/kubernetes/kubernetes/issues/117146

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: test/e2e/dra/test-driver/app/gomega.go
Comment (bart0sh) [reply test/e2e/dra/test-driver/app/gomega.go:38]: I don't think this is the case. Otherwise we'd need to do this for every loop in the code.

Comment (TommyStarK) [reply test/e2e/dra/test-driver/app/gomega.go:38]: Alright, I am misunderstanding then. @rsc you recently reported multiple issues related to what you called the loopvar sharing bug. Just for my understanding when do we fall into that situation ? Does it apply to every for range loop or is there specific criteria ?

Comment (pohly) [reply test/e2e/dra/test-driver/app/gomega.go:38]: It's not needed here because the loop variable is only used inside the loop body.

The loopvar problem occurs when the value is used after the loop is complete (referencing it in a goroutine, in a parallel test (which is ultimately a goroutine), storing a pointer to it, etc.).

Comment (TommyStarK) [reply test/e2e/dra/test-driver/app/gomega.go:38]: Thanks Ed and Patrick, it's crystal clear now :)

File: test/e2e_node/dra_test.go
Comment (pohly) [inline test/e2e_node/dra_test.go:113]: Can you add an explanation of how this test works? The key point will be why the pod remains in the pending state consistently.

**Review by pohly — CHANGES_REQUESTED** [review]

File: test/e2e/dra/test-driver/app/kubeletplugin.go
Comment (pohly) [reply test/e2e/dra/test-driver/app/kubeletplugin.go:154]: I wonder whether depending on timing could make this test flaky.

What if we optionally block until the call is guaranteed to have timed out? Something like:
```
if ex.timeOut {
   <-ctx.Done()
   return ctx.Err()
}   
```

Comment (bart0sh) [reply test/e2e/dra/test-driver/app/kubeletplugin.go]: Thank you for the review! Done.

File: test/e2e/dra/test-driver/app/kubeletplugin.go
Comment (bart0sh) [reply test/e2e/dra/test-driver/app/kubeletplugin.go]: done

File: test/e2e/dra/test-driver/app/kubeletplugin.go
Comment (pohly) [inline test/e2e/dra/test-driver/app/kubeletplugin.go:154]:
```suggestion
	// Block to emulate plugin stuckness or slowness.
	// By default the call will not be blocked as ex.block = false.
```

File: test/e2e/dra/test-driver/app/kubeletplugin.go
Comment (pohly) [inline test/e2e/dra/test-driver/app/kubeletplugin.go:221]: See above.

**Review by pohly — CHANGES_REQUESTED** [review]
One language nit, otherwise it looks good.

**Comment by pohly** [comment]
Please squash to make the PR ready for merging.

File: test/e2e/dra/test-driver/app/kubeletplugin.go
Comment (bart0sh) [reply test/e2e/dra/test-driver/app/kubeletplugin.go]: done

**Comment by bart0sh** [comment]
> Please squash to make the PR ready for merging.

The commits are independent and easy to understand. I fear that if I squash the commit would be less understandable. I can squash if you insist, though.

File: test/e2e/dra/test-driver/app/kubeletplugin.go
Comment (pohly) [inline test/e2e/dra/test-driver/app/kubeletplugin.go:154]:
```suggestion
	// Block to emulate plugin stuckness or slowness.
	// By default the call will not be blocked as ex.block = false.
```

File: test/e2e/dra/test-driver/app/kubeletplugin.go
Comment (pohly) [inline test/e2e/dra/test-driver/app/kubeletplugin.go:221]:
```suggestion
	// Block to emulate plugin stuckness or slowness.
	// By default the call will not be blocked as ex.block = false.
```

**Review by pohly — APPROVED** [review]
Looks like you lost the fix for the spelling nit when reorganizing the commits.

Except for that the commits are okay now. When I made that comment about "squashing", the commits weren't clean and squashing them into one would have resolved that.

I'll let you decide whether you want do fix this, feel free to lift the hold.

/lgtm
/approve
/hold

File: test/e2e/dra/test-driver/app/kubeletplugin.go
Comment (bart0sh) [reply test/e2e/dra/test-driver/app/kubeletplugin.go]: Ah, I missed the dots.  Thanks for pointing out. done.

**Comment by bart0sh** [comment]
@pohly now it should be ready to merge. Do we need to lgtm or it's enough to unhold?

**Review by pohly — APPROVED** [review]
/lgtm
/unhold

**Pull request merged**
